src/index.ts

```diff
@@ -1,61 +1,148 @@
     super(app, options, id)
-    this.var1 = options?.var1 || 'initial value'
-    // ...
+    this.secret = options?.secret || undefined
   }
 
   onStart(): void {
-    // If you need to do something specific on start, otherwise remove this function
-  }
-
-  onStop(): void {
-    // If you need to do something specific on stop, otherwise remove this function
+    if (Object.keys(this.eventConfigurations).length) {
+      this.webhookUrl = (this.configOptions.webhookPath || DEFAULT_WEBHOOK_PATH)
+      this.app.getLogger().info(`Registering Wix webhook: ${this.webhookUrl}`)
+      this.app.registerHTTPDelegate(`${this.webhookUrl}/provision`, this)
+      this.app.registerHTTPDelegate(`${this.webhookUrl}/:context`, this)
+      this.app.registerHTTPDelegate(`${this.webhookUrl}/:context/:action`, this)
+    }
   }
 
-  // Your events
   on(
-    options: _CONNECTOR_NAME_ConnectorEventOptions,
-    handler: any,
-    eventId: string,
+    options: WixConnectorEventOptions,
+    handler: (event: WixEvent, app: Reshuffle) => void,
+    eventId?: string,
   ): EventConfiguration {
-    if (!eventId) {
-      eventId = `_CONNECTOR_NAME_/${options.option1}/${this.id}`
-    }
-    const event = new EventConfiguration(eventId, this, options)
+    const event = new EventConfiguration(
+      eventId || `WixDataConnector/${options.action}/${this.id}`,
+      this,
+      options,
+    )
     this.eventConfigurations[event.id] = event
+    this.app.when(event, handler as any)
+    return event
+  }
 
-    this.app.when(event, handler)
+  onRemoveEvent(ec: EventConfiguration): void {
+    delete this.eventConfigurations[ec.id]
+  }
 
-    return event
+  async handle(req: Request, res: Response): Promise<boolean> {
+    if (this.started) {
+      await this.handleWebhookEvent({req, res, url: req.originalUrl})
+    } else {
+      res.json({"message": "Connector not configured"}).status(400)
+    }
+    return true
   }
 
-  // Your actions
-  action1(bar: string): void {
-    // Your implementation here
+  private async handleWebhookEvent(event: Record<string, any>) {
+    const path: string = this.extractAction(event.url)
+    const incoming = event.req.body
+    const ev: WixEvent = {
+      action: path as WixAction,
+      requestContext: incoming.requestContext,
+      collectionName: incoming.collectionName,
+      filter: incoming.filter,
+      sort: incoming.sort,
+      skip: incoming.skip,
+      limit: incoming.limit,
+      itemId: incoming.itemId,
+      item: incoming.item,
+      request: event.req,
+      response: event.res
+    }
+    if (!this.secret ||
+      (ev.requestContext.settings.secret &&
+        this.secret === ev.requestContext.settings.secret)) {
+
+      for (const ec of Object.values(this.eventConfigurations)) {
+        const storeAction = ec.options.action
+        const incoming = ev.action
+        if (storeAction == incoming) {
+          await this.app.handleEvent(ec.id, ev)
+          return
```

Comment:
you can have several connector.on listening to the same thing so I'll remove return to invoke them all

Reply:
But I want to handle the case where the connector is not configured for a specific event without a flag...